nuntium/tests/api/message_resource_test.py

```diff
@@ -318,3 +318,36 @@ def test_including_a_non_email_in_the_author_email(self):
 
         self.assertEquals(response.status_code, 400)
         self.assertFalse(Message.objects.filter(author_name='Felipipoo'))
+
+    def test_get_list_of_messages_filtered_by_popolo_uri(self):
+        popolo_uri = 'http://example.com/popolo.json#person-123'
+        # Create a person with a popolo identifier
+        person = PopoloPerson.objects.create(name='Test')
+        person.identifiers.create(scheme='popolo_uri', identifier=popolo_uri)
+        # Create an extra message that we don't expect to see in the response
+        message = Message.objects.create(
+            content=u'Test Content',
+            author_name=u'Felipe',
+            author_email=u"falvarez@votainteligente.cl",
+            subject=u'Fiera es una perra feroz',
+            writeitinstance=self.writeitinstance,
+            persons=[person],
+            )
+        Confirmation.objects.create(message=message)
+        message.recently_confirmated()
+
+        # Fetch a filtered list of message from the API
+        url = '/api/v1/message/'
+        params = self.data.copy()
+        params.update({
+            'person__popolo_uri': popolo_uri,
+        })
+        response = self.api_client.get(url, data=params)
+        self.assertValidJSONResponse(response)
+
+        expected_messages = Message.public_objects.filter(writeitinstance__in=self.user.writeitinstances.all(), person=person)
+        messages = self.deserialize(response)['objects']
+        self.assertEqual(len(messages), expected_messages.count())
```

Comment:
I think `expected_messages.count()` is always going to be 1 here, so I would probably change this to:
```python
        expected_message = Message.public_objects.get(writeitinstance__in=self.user.writeitinstances.all(), person=person)
        messages_from_api = self.deserialize(response)['objects']
        self.assertEqual(len(messages_from_api), 1)
        message_from_api = messages_from_api[0]
        self.assertEqual(message_from_api['id'], expected_message.id)
        self.assertEqual(len(message_from_api['people']), 1)
        self.assertEqual(message_from_api['people'][0]['id'], person.id)
```

... which I think is easier to read. (Your version would be easier to extend if one wants to add other example message in the future, say, but I think I value the readability more. This also checks that it's the right message being returned.)

Reply:
Yes, this is much better than my version, thanks! I've made this change, and the two above, in b703d5bade13553c8b321eede18c70f3fbcccd46.